core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -1006,6 +1005,16 @@ private void cleanupExpiredRemoteLogSegments() throws RemoteStorageException, Ex
 
             // Update log start offset with the computed value after retention cleanup is done
             remoteLogRetentionHandler.logStartOffset.ifPresent(offset -> handleLogStartOffsetUpdate(topicIdPartition.topicPartition(), offset));
```

Comment:
we might have a situation where we don't delete the segments because we are not the leader anymore but we still update the log start offset. It is ok to do that. Please add a comment about it here. For example:
1. we update log start offset
2. replica fetch takes place and completes. 
3. replica updates its log start offset based on our new log start offset
4. that replica becomes new leader
5. we skip deleting the files here because we are not the leader anymore
6. the new leader will still delete these files but it will delete them under the delete everything below log start offset case.